### PR TITLE
Merge 0.21.12

### PR DESCRIPTION
**Pull request by rossabaker**
This had a non-trivial port of #3920.  I'd like to give @RafalSumislawski, @ashwinbhaskar, and @hamnis a chance to double check the URI part before we release 1.0.0-M8.

### CODE REVIEW

**Comment by ashwinbhaskar** [comment]
there is a failing test on async http client `strip fragments from URI (11 ms)`

File: website/src/hugo/content/changelog.md [hunk 1]
```diff
@@ -8,6 +8,31 @@ Maintenance branches are merged before each new release. This change log is
...
+## Bugfixes
+
+### http4s-core
+
+* [#3911](https://github.com/http4s/http4s/pull/3911): Support raw query strings. Formerly, all query strings were stored as a vector of key-value pairs, which was lossy in the percent-encoding of sub-delimiter characters (e.g., '+' vs '%2B').  Queries constructed with `.fromString` will be rendered as-is, for APIs that assign special meaning to sub-delimiters.
+* [#3921](https://github.com/http4s/http4s/pull/3921): Fix rendering of URIs with colons. This was a regression in v0.21.9.
+
+### http4s-circe
+
+* [#3906](https://github.com/http4s/http4s/pull/3906): Fix streamed encoder for empty stream. It was not rendering the `{`.
```
Comment (hamnis) [inline website/src/hugo/content/changelog.md:22]: wrong brace? Should be `[`

**Comment by RafalSumislawski** [comment]
The backport of #3920 looks good to me.

**Review by ashwinbhaskar — APPROVED** [review]

**Comment by rossabaker** [comment]
> there is a failing test error in async http client ClientRouteTestBattery

Yeah, that's one of the flaky ones.  I don't know whether we've hit a tipping point of if GitHub Actions changed, but it has gotten worse.

**Comment by rossabaker** [comment]
GitHub Actions has gone out for a walk, but most of the builds are fine.

**Pull request merged**